File: wallendar-app/app/src/main/java/com/td/wallendar/repositories/interfaces/ChargesRepository.java
package com.td.wallendar.repositories.interfaces;

import com.td.wallendar.dtos.request.AddChargeRequest;
import com.td.wallendar.dtos.response.AddChargeResponse;
import com.td.wallendar.models.Charge;

import io.reactivex.Observable;

public interface ChargesRepository {
    Observable<AddChargeResponse> addCharge(final long groupId, final AddChargeRequest addChargeRequest);
}

File: wallendar-app/app/src/main/java/com/td/wallendar/repositories/interfaces/ChargesRepository.kt
package com.td.wallendar.repositories.interfaces

import com.td.wallendar.dtos.request.AddChargeRequest
import com.td.wallendar.models.Charge
import io.reactivex.Single

interface ChargesRepository {
    fun addCharge(groupId: Long, addChargeRequest: AddChargeRequest, isEvent: Boolean): Single<Charge>
}